spec/rubocop/cop/style/negated_if_spec.rb

```diff
@@ -1,90 +1,207 @@
+      inspect_source(cop,
+                     ['if !a_condition',
+                      '  some_method',
+                      'end',
+                      'some_method if !a_condition'])
+      expect(cop.messages).to eq(
+        ['Favor `unless` over `if` for negative ' \
+         'conditions.'] * 2
+      )
+    end
+
+    it 'registers an offense for unless with exclamation point condition' do
+      inspect_source(cop,
+                     ['unless !a_condition',
+                      '  some_method',
+                      'end',
+                      'some_method unless !a_condition'])
+      expect(cop.messages).to eq(['Favor `if` over `unless` for negative ' \
+                                  'conditions.'] * 2)
+    end
+
+    it 'registers an offense for if with "not" condition' do
+      inspect_source(cop,
+                     ['if not a_condition',
+                      '  some_method',
+                      'end',
+                      'some_method if not a_condition'])
+      expect(cop.messages).to eq(
+        ['Favor `unless` over `if` for negative ' \
+         'conditions.'] * 2
+      )
+      expect(cop.offenses.map(&:line)).to eq([1, 4])
+    end
+
+    it 'accepts an if/else with negative condition' do
+      inspect_source(cop,
+                     ['if !a_condition',
+                      '  some_method',
+                      'else',
+                      '  something_else',
+                      'end',
+                      'if not a_condition',
+                      '  some_method',
+                      'elsif other_condition',
+                      '  something_else',
+                      'end'])
+      expect(cop.offenses).to be_empty
+    end
+
+    it 'accepts an if where only part of the condition is negated' do
+      inspect_source(cop,
+                     ['if !condition && another_condition',
+                      '  some_method',
+                      'end',
+                      'if not condition or another_condition',
+                      '  some_method',
+                      'end',
+                      'some_method if not condition or another_condition'])
+      expect(cop.offenses).to be_empty
+    end
+
+    it 'accepts an if where the condition is doubly negated' do
+      inspect_source(cop,
+                     ['if !!condition',
+                      '  some_method',
+                      'end',
+                      'some_method if !!condition'])
+      expect(cop.offenses).to be_empty
+    end
+
+    it 'is not confused by negated elsif' do
+      inspect_source(cop,
+                     ['if test.is_a?(String)',
+                      '  3',
+                      'elsif test.is_a?(Array)',
+                      '  2',
+                      'elsif !test.nil?',
+                      '  1',
+                      'end'])
+
+      expect(cop.offenses).to be_empty
+    end
+
+    it 'autocorrects by replacing if not with unless' do
```

Comment:
I'm not sure what you mean. Do you mean the three autocorrect tests at the end? Or all of the tests inside of the “both” block? Or do you mean it doesn't seem to be testing the “both” style because we're not specifically setting the `EnforcedStyle` configuration to “both”?

What I did was I moved all the existing tests into the “both” block, and then added new tests specifically for “prefix” and “postfix”. My thinking was since “prefix” and “postfix” styles didn't previously exist, all the existing tests were inherently testing the “both” style.